refund_users.ts

```diff
@@ -13,41 +14,93 @@ interface Users {
     mangoAccountPks: string,
 }
 
-export async function main(users: Users[]) {
+enum Result {
+    SUCCESS = 0,
+    FAILURE = 1
+};
+
+
+export async function main(users: Users[],
+                           authority: Keypair,
+                           targetBalance: number,
+                           n_try: number) {
     // cluster should be in 'devnet' | 'mainnet' | 'localnet' | 'testnet'  
     const endpoint = process.env.ENDPOINT_URL || 'http://localhost:8899';
     const connection = new Connection(endpoint, 'confirmed');
-    const authority = Keypair.fromSecretKey(
-        Uint8Array.from(
-          JSON.parse(
-            process.env.KEYPAIR ||
-              readFileSync('authority.json', 'utf-8'),
-          ),
-        ),
-      );
-    let promises : Promise<String>[]= []
-    let blockHash = await connection.getLatestBlockhash();
-    for (const user of users) {
-        let userPubkey = new web3.PublicKey(user.publicKey)
-        const ix = SystemProgram.transfer({
-            fromPubkey: authority.publicKey,
-            lamports: LAMPORTS_PER_SOL,
-            toPubkey: userPubkey,
-        })
-        let tx = new web3.Transaction().add(ix);
-        tx.recentBlockhash = blockHash.blockhash;
-        promises.push(connection.sendTransaction(tx, [authority]))
+
+    let accounts_to_fund = new Set<[Users, number]>();
```

Comment:
why set could be a simple array.

Reply:
I've addressed this thing and also added one commit to not stop execution in case of send_txs error (retrying in 2s anyways)